elytron/WFLY-16306-keycloak-saml-feature-pack.adoc

```diff
@@ -0,0 +1,331 @@
+that need to propagate their identities to EJBs as well.
+
+The CLI scripts that would have been added via the manual installation steps won't be added to the resulting installation
+since the required resources will be added already during provisioning.
+
+Note that Keycloak and WildFly have different release cadences. This means anyone using Galleon to keep their WildFly
+installation up to date should be able to receive updates to the core WildFly installation independently of the new
+Keycloak feature pack. Having the new Keycloak feature pack installed should not prevent the server from being updated
+once later WildFly versions are available. Similarly, WildFly releases should not trigger the need for an update to Keycloak.
+
+==== Bare Metal vs. Cloud Use Cases
+
+For the bare metal use case, the `keycloak-client-saml` layer should be used.
+
+For WildFly s2i, when using the automatic registration of SAML deployments feature, the `keycloak-saml` layer should be used.
+(For this use case, the necessary `elytron`, `undertow`, and `ejb` subsystem configuration will be handled via a CLI script
+that is generated at boot time.)
+
+Otherwise, for WildFly s2i with manual deployments (i.e., with `keycloak-saml` configuration handled via a CLI script or included
+within the deployment itself), the `keycloak-client-saml` layer should be used.
+
+For bootable JAR on the cloud, the `keycloak-client-saml` layer should be used.
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+
+Published artifacts will be available from online maven repositories only. No offline maven repository support will be provided.
+
+The feature pack and resulting layer being provisioned are for securing web applications accessed directly via HTTP.
+This feature request does not extend to securing management interfaces, Remoting invocations, or EJB over HTTP invocations.
+
+== Examples
+
+This section contains examples of how the new feature pack will be installed using different approaches.
+
+=== Provisioned using the Galleon CLI
+
+A server installation can be provisioned using the Galleon CLI which would need to be downloaded and installed first.
+
+Start the CLI:
+
+----
+java -jar galleon-cli-5.0.6.Final.jar
+----
+
+First provision a WildFly server with the `web-server` later:
+
+----
+install wildfly:current --layers=web-server --dir=/home/fjuma/tmp/galleon/wildfly
+----
+
+Then install the Keycloak Elytron SAML adapter:
+
+----
+install org.keycloak:keycloak-saml-adapter-galleon-pack:999-SNAPSHOT --layers=keycloak-client-saml --dir=/home/fjuma/tmp/galleon/wildfly
+----
+
+=== Provisioned using org.wildfly.plugins:wildfly-maven-plugin.
+
+The following is an example of using the `wildfly-maven-plugin` to provision a server containing the Keycloak
+SAML subsystem.
+
+[source,xml]
+----
+<plugin>
+    <groupId>org.wildfly.plugins</groupId>
+    <artifactId>wildfly-maven-plugin</artifactId>
+    <version>4.0.0.Final</version>
+    <configuration>
```

Comment:
@fjuma shouldn't we run some `configure-saml.cli` cli script here in order to point the EAP instance to RHSSO? I mean at some point we have to add something like `http://127.0.0.1:8180/auth/realms/my-realm/protocol/saml` to the EAP configuration ....

Reply:
The configuration can either be specified within the deployment itself using a `keycloak-saml.xml` file or a CLI script can be run to add the configuration to the `keycloak-saml` subsystem. For both of these cases, it's possible to get the necessary configuration from the RH SSO console from the Installation tab.